core/src/streams.rs

```diff
@@ -393,14 +402,126 @@ impl<'gc> NetStream<'gc> {
                     >= write.preload_offset;
 
                 match tag.data {
-                    FlvTagData::Audio(FlvAudioData { .. }) => {
-                        tracing::warn!("Stub: Stream audio processing");
+                    FlvTagData::Audio(FlvAudioData {
+                        format,
+                        rate,
+                        size,
+                        sound_type,
+                        data,
+                    }) => {
+                        let mut substream = match &mut write.stream_type {
+                            Some(NetStreamType::Flv {
+                                audio_stream: Some((substream, audio_handle)),
+                                ..
+                            }) if context.audio.is_sound_playing(*audio_handle) => {
+                                substream.clone()
+                            }
+                            Some(NetStreamType::Flv {
+                                audio_stream, //None or not playing
+                                ..
+                            }) => {
+                                let substream = Substream::new(slice.buffer().clone());
+                                let audio_handle = (|| {
```

Comment:
This might be a technicality that I'm unaware of, but asking anyway out of curiosity: What's the point of an instantly called lambda here, as compared to a simple block?

Reply:
I wanted to use `?` in a function that does not return a `Result`. Didn't know you could have blocks that return like functions, I'll try that next time.

AFAIK this code is gone now so I'm just going to mark this as resolved.

Reply:
> I wanted to use ? in a function that does not return a Result. Didn't know you could have blocks that return like functions, I'll try that next time.

Turns out, it's not a `return`, but a `break`. And it's not that old: https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html#break-from-labeled-blocks

> AFAIK this code is gone now so I'm just going to mark this as resolved.

I'm going commit-by-commit, as I find it easier to understand the changes that way. Also I'm interested in how the commit history looks.